data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/client/OpenSearchAccessor.java

```diff
@@ -75,9 +93,37 @@ public CreatePointInTimeResponse createPit(final CreatePointInTimeRequest create
     }
 
     @Override
-    public SearchPitResponse searchWithPit(final SearchPitRequest searchPitRequest) {
-        // todo: implement
-        return null;
+    public SearchPointInTimeResponse searchWithPit(final SearchPointInTimeRequest searchPointInTimeRequest) {
+        try {
+            final SearchResponse<ObjectNode> searchResponse = openSearchClient.search(
+                    SearchRequest.of(builder -> {
+                        builder
+                            .pit(Pit.of(pitBuilder -> pitBuilder.id(searchPointInTimeRequest.getPitId()).keepAlive(searchPointInTimeRequest.getKeepAlive())))
+                            .size(searchPointInTimeRequest.getPaginationSize())
+                            .sort(SortOptions.of(sortOptionsBuilder -> sortOptionsBuilder.doc(ScoreSort.of(scoreSort -> scoreSort.order(SortOrder.Asc)))))
+                            .query(Query.of(query -> query.matchAll(MatchAllQuery.of(matchAllQuery -> matchAllQuery.queryName("*")))));
+
+                        if (Objects.nonNull(searchPointInTimeRequest.getSearchAfter())) {
+                            builder.searchAfter(searchPointInTimeRequest.getSearchAfter());
+                        }
+
+                        return builder;
+        }), ObjectNode.class);
+
+            final List<Event> documents = searchResponse.hits().hits().stream()
+                    .map(hit -> JacksonEvent.builder()
+                            .withData(hit.source())
+                            .withEventMetadataAttributes(Map.of(DOCUMENT_ID_METADATA_ATTRIBUTE_NAME, hit.id(), INDEX_METADATA_ATTRIBUTE_NAME, hit.index()))
+                            .withEventType(EventType.DOCUMENT.toString()).build())
+                    .collect(Collectors.toList());
+
+            return SearchPointInTimeResponse.builder()
+                    .withDocuments(documents)
+                    .withNextSearchAfter(searchResponse.hits().hits().get(searchResponse.hits().hits().size() - 1).sort())
```

Comment:
I'm not sure I follow how you get the next search after. Is there no order from the OpenSearch response that could help here?

Also, you have an inline `sort()`. That should probably be pulled out of this line as it is somewhat signficiant.

Reply:
I see, this is the OpenSearch "sort" model, not a collection sort.